### DISCUSSION TITLE
Contract instance throws error TypeError: Cannot read properties of undefined (reading 'fee')

### DISCUSSION THREAD

**Discussion by velvety-cc** [wevm/viem#1494] [Question]
First time using Viem, and I'm just playing with the contract instances. 

When I try to read a simple Uniswap v3 pool fee:

```
const contract = getContract({
    address: poolAddr,
    abi: poolAbi,
    client,
});

const fee = await contract.read.fee();
```

It throws TypeError: Cannot read properties of undefined (reading 'fee').

However, it works fine when I use individual calls:
```
const fee = await client.readContract({
    address: poolAddr,
    abi: poolAbi,
    functionName: 'fee',
});
```

Anybody knows what could be wrong here?

**Comment by travs** [comment]
@velvety-cc did you ever find a solution to this?

I'm facing the same issue. In my case, I am using `viem@2.0.3` with hardhat (`@nomicfoundation/hardhat-viem@1.0.2`).

**Reply by tmm** [reply]
Can you share a minimal reproduction?

**Reply by travs** [reply]
Yes, though I was able to resolve in the meantime.

Minimal reproduction: https://github.com/travs/viem-issue-reproduction.git

The fix for now seems to sadly be to install a different viem version (`viem@1`).

Not sure what the actual issue was in the end, since `read` property does not seem to exist on the smart contract given error message, but that property seems to exist in viem 1 and 2.

Using the `--legacy-peer-deps` flag should have indicated this to me earlier, and perhaps the real fix is for hardhat to [support viem version 2](https://github.com/NomicFoundation/hardhat/issues/4749).

**Comment by LeoDavidov** [comment]
Same issue...

**Reply by jxom** [reply]
Commenting "same issue" doesn't help. Minimal reproduction please :)

**Reply by LeoDavidov** [reply]
Here it is: https://github.com/LeoDavidov/viem-mre
Please let me know if it happens because I did something wrong

**Reply by jxom** [reply]
```diff
import {createPublicClient, getContract, webSocket} from "viem";
import {base} from "viem/chains";


const publicClient = createPublicClient({
    chain: base,
    transport: webSocket('ws_url')
})

const wethContract = getContract({
    address: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
    abi: [
        {
            "constant": true,
            "inputs": [
                {
                    "name": "_owner",
                    "type": "address"
                }
            ],
            "name": "balanceOf",
            "outputs": [
                {
                    "name": "balance",
                    "type": "uint256"
                }
            ],
            "payable": false,
            "type": "function"
        }
    ],
-   publicClient,
+   client: publicClient,
})

const balance = await wethContract.read.balanceOf(['0x2d2f0abd526e87ca05eb19558d3d5fa8460266da4d35ef19d616d2e6ef16bfb0']);
```

**Reply by LeoDavidov** [reply]
The issue still persists after this adjustment

**Reply by LeoDavidov** [reply]
I have fixed the code and it works now. Can't exactly specify what was the issue but I switched to Ethereum network and it works. I think the issue lied in the fact that I was using ethereum network ws link and accessing a base network coin
